### PR TITLE
Fixed Bad Config Preventing Startup

### LINKED ISSUE #907: Bad Garden Config stops startup
If a Garden Connection Info is not populated for STOMP, but that is the set Connect Type, it will stop the Startup and not allow it to proceed. Need to better handle failed STOMP startups.

### PR DESCRIPTION
**Pull request by west270**
Fixed error that cause Beer Garden not to start when garden is configured to use stomp but is the field in config are empty. Also added logger warning message for connection failures to include a list of gardens affected so if the problem is with the connection config you can find which gardens to check in the logs.

Fixes #907

### CODE REVIEW

**Review by TheBurchLog — COMMENTED** [review]
The fact that it attempts to connect with missing configurations is bad. We need logic added in the UI that when the Connection Type of STOMP is selected they are required to populate the fields.

File: src/app/beer_garden/api/stomp/server.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def __init__(
         username=None,
         password=None,
     ):
+        self.host_and_port = host_and_ports
```
Comment (hazmat345) [inline src/app/beer_garden/api/stomp/server.py:110]: Shouldn't these be named the same?

File: src/app/beer_garden/api/stomp/server.py [hunk 2]
```diff
@@ -125,34 +126,44 @@ def __init__(
...
-                        destination=self.subscribe_destination,
-                        id=self.username,
-                        ack="auto",
-                        headers={
-                            "subscription-type": "MULTICAST",
-                            "durable-subscription-name": self.subscribe_destination,
-                        },
+        if (
+            self.host_and_port[0][0]
+            and self.host_and_port[0][1]
```
Comment (hazmat345) [inline src/app/beer_garden/api/stomp/server.py:133]: Can't `host_and_port[0]` be `None`? If that happens this is going to blow up.

**Review by hazmat345 — CHANGES_REQUESTED** [review]

File: src/app/beer_garden/api/stomp/server.py (hunk 2 above)
Comment (west270) [reply src/app/beer_garden/api/stomp/server.py:133]: @hazmat345 `host_and_port` defaults to `None` if nothing is passed in, but a value is always passed in. If the value of `host` and `port` are `None` then value passed in on init is ```host_and_port = [(None, None)]```, so `host_and_port` should not be `None`.

File: src/app/beer_garden/api/stomp/server.py [hunk 3]
```diff
@@ -128,42 +128,43 @@ def __init__(
...
+                                    "durable-subscription-name": self.subscribe_destination,
+                                },
+                            )
+                        if connected_message is not None and self.conn.is_connected():
+                            logger.info("Stomp successfully " + connected_message)
+
+                    except Exception as e:
+                        logger.warning(str(e))
+                        logger.warning(
+                            f"Affected gardens are {[garden.get('name') for garden in gardens]}"
```
Comment (hazmat345) [inline src/app/beer_garden/api/stomp/server.py:161]: If you separate these into 2 calls there could be any number of lines in between them in the actual log file. It's safer to combine them into one call:

```python
logger.warning("Error connecting: {e}. Affected gardens are {[garden.get('name') for garden in gardens]}")
```

File: src/app/beer_garden/api/stomp/server.py [hunk 4]
```diff
@@ -128,42 +128,43 @@ def __init__(
...
+                        )
+                        logger.warning(
```
Comment (hazmat345) [inline src/app/beer_garden/api/stomp/server.py:163]: This should be debug level.

**Review by hazmat345 — CHANGES_REQUESTED** [review]

**Review by hazmat345 — APPROVED** [review]
LGTM, 👍🏼

**Pull request merged**